core/ecschema-editing/src/test/Editing/Exception.test.ts

```diff
@@ -34,12 +34,46 @@ describe("SchemaEditingError tests", () => {
     ];
     const innerError = new SchemaEditingError(ECEditingStatus.RuleViolation, new SchemaId(testKey), undefined, diagnostics);
     const error = new SchemaEditingError(ECEditingStatus.AddSchemaReference, new SchemaId(testKey), innerError);
-    expect(error.toDebugString()).to.equal(`ECEditingStatus.AddSchemaReference: Inner error: ECEditingStatus.RuleViolation: Rule violations occurred from Schema testSchema: ${getRuleViolationMessage(diagnostics)}`);
+    expect(error.toDebugString()).to.equal(`ECEditingStatus.AddSchemaReference: While performing task '${ECEditingStatus[error.errorNumber]}' an error occurred editing Schema ${testKey.name}. Inner error: ECEditingStatus.RuleViolation: Rule violations occurred from Schema TestSchema: ${getRuleViolationMessage(diagnostics)}`);
   });
 
   it("Should create proper debug string with inner exception of type Error", async () => {
     const innerError = new Error("Could not create EntityClass testEntityClass for some unknown reason.");
-    const error = new SchemaEditingError(ECEditingStatus.CreateSchemaItemFailed, new SchemaItemId(SchemaItemType.EntityClass, "TestEntity", testKey), innerError);
-    expect(error.toDebugString()).to.equal(`ECEditingStatus.CreateSchemaItemFailed: Inner error: Could not create EntityClass testEntityClass for some unknown reason.`);
+    const error = new SchemaEditingError(ECEditingStatus.CreateSchemaItemFailed, new ClassId(SchemaItemType.EntityClass, "TestEntity", testKey), innerError);
+    expect(error.toDebugString()).to.equal(`ECEditingStatus.CreateSchemaItemFailed: While performing task '${ECEditingStatus[error.errorNumber]}' an error occurred editing Class TestSchema.TestEntity. Inner error: Could not create EntityClass testEntityClass for some unknown reason.`);
   });
+
+  it("Should create correct error message for Schema task error", async () => {
+    const identifier = new SchemaId(testKey);
+    const error = new SchemaEditingError(ECEditingStatus.CreateElement, identifier, new Error("inner error"));
+    expect(error.message).to.equal(`While performing task '${ECEditingStatus[error.errorNumber]}' an error occurred editing Schema ${identifier.name}.`);
+  });
+
+  it("Should create correct error message for SchemaItem task error", async () => {
+    const identifier = new SchemaItemId(SchemaItemType.Unit, "TestUnit", testKey);
+    const error = new SchemaEditingError(ECEditingStatus.CreateSchemaItemFailed, identifier, new Error("inner error"));
+    expect(error.message).to.equal(`While performing task '${ECEditingStatus[error.errorNumber]}' an error occurred editing SchemaItem ${identifier.name}.`);
```

Comment:
I know we've been doing it that way for a while now, but I think, we should get in the habit of having static error messages. In this case
```ts
"While performing task 'CreateSchemaItemFailed' an error occurred editing SchemaItem TestUnit."
```
That's a little more effort if we change something, but you see at once what the expected result is, and things don't slip through so easily if we change something on the backend side than if we'd compose the expected results dynamically.

Reply:
I think we should discuss this.  I want to make sure I understand what you are saying.